src/libcollections/vec.rs

```diff
@@ -1301,6 +1301,41 @@ impl<T: Clone> Vec<T> {
     }
 }
 
+impl<T: Default> Vec<T> {
```

Comment:
Just a thought, separate `impl` blocks for different bounds seems archaic (from the time before `where` clauses), so we probably don't want to continue that.